strawberryfields/compilers/__init__.py

```diff
@@ -31,39 +31,30 @@
 circuit that belongs in the target circuit class.
 
 The circuit class database :attr:`circuit_db` is a dictionary mapping the circuit family
-short name to the corresponding CircuitSpecs instance.
+short name to the corresponding Compiler instance.
 In particular, for each backend supported by Strawberry Fields the database contains a
-corresponding CircuitSpecs instance with the same short name, used to validate Programs to be
+corresponding Compiler instance with the same short name, used to validate Programs to be
 executed on that backend.
 """
-from .circuit_specs import CircuitSpecs, Ranges
-from .X8 import X8Specs, X8_01
-from .X12 import X12Specs, X12_01, X12_02
+from .compiler import Compiler, Ranges
 from .xcov import Xcov
 from .xunitary import Xunitary
-from .fock import FockSpecs
-from .gaussian import GaussianSpecs
-from .gbs import GBSSpecs
-from .tensorflow import TFSpecs
+from .fock import Fock
+from .gaussian import Gaussian
+from .gbs import GBS
 from .gaussian_unitary import GaussianUnitary
 
-specs = (
-    X8Specs,
-    X8_01,
-    X12Specs,
-    X12_01,
-    X12_02,
-    FockSpecs,
-    GaussianSpecs,
-    GBSSpecs,
-    TFSpecs,
```

Comment:
Is compilation for the TFBackend something that is no longer relevant?

Reply:
Yep, the TF backend simply uses the `Fock` compiler. Previously, we required that each backend had its own CircuitSpec, but with the refactor, different backends can now use the same compiler.

Reply:
got it! How does it use the `Fock` compiler? :thinking: Would a `compiler = "fock"` be needed for the TFBackend?

Reply:
Yes, I think it is added as a class attribute in the backends! Note this is not a new change, the TFBackend previously specified `circuit_spec = "tf"`. But as noted above, this is entirely redundant, since the TF and Fock backends support the same gateset.

Reply:
So, the `Fock` compiler is set to default (`compiler = "Fock"`) in the `BaseFock` class inside `backends.base.py`. The `TFBackend` inherits from `BaseFock` and doesn't replace the `compiler`. :slightly_smiling_face:

Reply:
thanks guys, these were the missing pieces :)

Reply:
oh that's even nicer!! I did that a couple of weeks ago in this PR, and completely forgot lol